runtime/gc_glue_java/ArrayletObjectModel.cpp

```diff
@@ -43,13 +43,21 @@ GC_ArrayletObjectModel::AssertBadElementSize()
 	Assert_MM_unreachable();
 }
 
-#if defined(J9VM_GC_ENABLE_DOUBLE_MAP)
 void
-GC_ArrayletObjectModel::AssertNotEmptyArrayletLeaves(UDATA sizeInElements, UDATA arrayletLeafCount)
+GC_ArrayletObjectModel::AssertArrayletIsDiscontiguous(J9IndexableObject *objPtr)
 {
-	Assert_MM_true((0 == sizeInElements) || (arrayletLeafCount > 0));
-}
+#if defined(J9VM_GC_ENABLE_DOUBLE_MAP)
+	if (!isDoubleMappingEnabled())
 #endif /* J9VM_GC_ENABLE_DOUBLE_MAP */
+	{
+		MM_GCExtensionsBase* extensions = MM_GCExtensionsBase::getExtensions(_omrVM);
+		UDATA arrayletLeafSize = _omrVM->_arrayletLeafSize;
+		UDATA remainderBytes = getDataSizeInBytes(objPtr) % arrayletLeafSize;
+		if (0 != remainderBytes) {
+			Assert_MM_true((getSpineSize(objPtr) + remainderBytes + extensions->getObjectAlignmentInBytes()) > arrayletLeafSize);
```

Comment:
Would you please add comment here to explain that with exception of Double Mapping request the only case when layout Discontiguous can be selected is Arraylet Hybrid representation does not fit a region? It would improve readability. My original confusion when I read this first time was "Discontiguous" in function name refer to Discontiguous Layout of Arraylet, not Discontiguous Array representation.

Reply:
I believe change can be merge without this suggested comment. We can add it later if necessary

Reply:
Sorry, I completely missed this comment. Let me know if you want the comment in this PR and I can insert it.